mercantile/__init__.py

```diff
@@ -133,7 +133,20 @@ def _parse_tile_arg(*args):
     if len(args) == 1:
         args = args[0]
     if len(args) == 3:
-        return Tile(*args)
+        tile = Tile(*args)
+
+        xtile, ytile, zoom = tile
+
+        if zoom != int(zoom) or not int(zoom) >= 0:
+            raise TileArgParsingError("zoom must be a positive integer")
```

Comment:
How about `InvalidZoomError` instead? `TileArgParsingError` is intended for unacceptable number of arguments (anything other than 1 or 3).